src/c++/perf_analyzer/genai-perf/genai_perf/export_data/json_exporter.py

```diff
@@ -26,29 +26,33 @@
 
 
 import json
-from argparse import Namespace
 from enum import Enum
-from pathlib import Path
 from typing import Dict
 
-from genai_perf.constants import DEFAULT_OUTPUT_DATA_JSON
+import genai_perf.logging as logging
+
+DEFAULT_OUTPUT_DATA_JSON = "profile_export_genai_perf.json"
+
+logger = logging.getLogger(__name__)
 
 
 class JsonExporter:
     """
     A class to export the statistics and arg values in a json format.
     """
 
-    def __init__(self, stats: Dict, args: Namespace, extra_inputs: Dict):
-        self._stats = stats
-        self._args = dict(vars(args))
-        self._extra_inputs = extra_inputs
+    def __init__(self, config: Dict):
+        self._stats: Dict = config["stats"]
+        self._args = dict(vars(config["args"]))
+        self._extra_inputs = config["extra_inputs"]
+        self._output_dir = config["artifact_dir"]
```

Comment:
This is scary to me. A generic dict passed in with 4 hardcoded key names. That is a lot of knowledge that this class needs to have, and there is no built in checking for typos. Does it make sense to create a data class to hide some of these details?

Reply:
config.stats <-- this will give a clear error if there is a typo. You can go to the definition of `config` to find all of the data members, and any potential documentation on the contents. This also allows typing checks.

config["stats"] <-- I have no way of tracing back where this came from or what else might be in this dictionary. It could come from anywhere and could contain anything. The caller and the consumer both have to have undocumented knowledge of the contents. A hidden agreement. If we later decide to change it from "stats" to "statistics", suddenly I'm doing a hopeful search and replace of a string across the code base.

Reply:
It looks like you made this change? If so this can be resolved

Reply:
yes. great suggestion